protocol/x/clob/keeper/liquidations.go

```diff
@@ -47,22 +47,42 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			return err
 		}
 
-		liquidationOrders = append(liquidationOrders, *liquidationOrder)
+		liquidationOrdersForSorting = append(liquidationOrdersForSorting, *liquidationOrder)
 	}
 
-	// Sort liquidation orders.
-	k.SortLiquidationOrders(ctx, liquidationOrders)
+	// Sort liquidation orders. The most underwater accounts should be liquidated first.
+	// These orders are only used for sorting. When we match these orders here in PrepareCheckState,
+	// liquidation matches will be put into the Operations Queue. However, when we process liquidations,
+	// we will generate a new liquidation order for each subaccount because previous liquidation orders
+	// can alter quantity sizes of subsequent liquidation orders.
+	k.SortLiquidationOrders(ctx, liquidationOrdersForSorting)
 
 	// Attempt to place each liquidation order and perform deleveraging if necessary.
 	numFilledLiquidations := uint32(0)
 	numAttemptedDeleveraging := uint32(0)
-	for i := 0; numFilledLiquidations < k.MaxLiquidationOrdersPerBlock && i < len(liquidationOrders); i++ {
-		optimisticallyFilledQuantums, _, err := k.PlacePerpetualLiquidation(ctx, liquidationOrders[i])
+	for i := 0; numFilledLiquidations < k.MaxLiquidationOrdersPerBlock && i < len(liquidationOrdersForSorting); i++ {
+		liquidationOrderSubaccountId := liquidationOrdersForSorting[i].GetSubaccountId()
+
+		// Generate a new liquidation order with the appropriate order size from the sorted subaccount ids.
+		liquidationOrder, err := k.MaybeGetLiquidationOrder(ctx, liquidationOrderSubaccountId)
+		if err != nil {
+			// Subaccount might not always be liquidatable if previous liquidation orders
+			// improves the net collateral of this subaccount.
+			if errors.Is(err, types.ErrSubaccountNotLiquidatable) {
+				telemetry.IncrCounter(1, metrics.MaybeGetLiquidationOrder, metrics.SubaccountsNotLiquidatable, metrics.Count)
```

Comment:
nit: use a diff metric here

Reply:
Going to remove this since the sorting stat should encompass what we want to report